### PR TITLE
MANOPD-76951 Fix "remote_node" procedure for miniHA

### PR DESCRIPTION
**Pull request by Corey-666**
### Description
Job fails if remove **shutdowned** node using maintenance procedure for **miniHA** scheme. When preserving of the information about the procedure the actual number of nodes is not taken into account.

### Solution
Additionally, use the **get_final_nodes()** function to get the actual list of nodes.

### How to apply
none

### Test Cases

**TestCase 1**

Test Configuration:
miniHA cluster of 3 master nodes

Steps:
1. Shutdown master node
2. Remove node from cluster using maintenance procedure
3. The node removal procedure will be fully executed, but the job will fail at the final stage of generating logs about the performed procedure. 

Results:

| Before | After |
| ------ | ------ |
| Job successfully finished | Job failure with python traceback error |

### Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Integration CI passed
- [ ] Unit tests. If Yes list of new/changed tests with brief description
- [ ] There is no merge conflicts

#### Unit tests
Indicate new or changed unit tests and what they do, if any.

### CODE REVIEW

File: kubemarine/core/utils.py [hunk 1]
```diff
@@ -342,7 +342,7 @@ def rotation_file(self):
                 for i, file in enumerate(files):
                     if i >= not_pack_file and i < delete_old:
                         if 'tar.gz' not in file:
-                            control_plane.sudo(f'tar -czvf {self.dir_path + file + ".tar.gz"} {self.dir_path + file} &&'
+                            control_plane.get_final_nodes().sudo(f'tar -czvf {self.dir_path + file + ".tar.gz"} {self.dir_path + file} &&'
```
Comment (ilia1243) [inline kubemarine/core/utils.py:345]: Here control_plane is a single node which is surely not a node to be removed. So the change is unnecessary.

File: kubemarine/core/utils.py [hunk 2]
```diff
@@ -386,7 +386,7 @@ def collect_info_all_control_plane(self):
         This method is used to transfer backup logs from the main control-plane to the new control-plane.
         """
 
-        node = self.cluster.nodes['control-plane'].get_initial_nodes().get_first_member(provide_node_configs=True)
+        node = self.cluster.nodes['control-plane'].get_final_nodes().get_initial_nodes().get_first_member(provide_node_configs=True)
```
Comment (ilia1243) [inline kubemarine/core/utils.py:389]: This method is invoked only during add_node procedure, so change is not necessary.

File: kubemarine/core/utils.py (hunk 1 above)
Comment (Corey-666) [reply kubemarine/core/utils.py]: Fixed

File: kubemarine/core/utils.py (hunk 2 above)
Comment (Corey-666) [reply kubemarine/core/utils.py]: fixed too

**Review by ilia1243 — APPROVED** [review]

**Pull request merged**